src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -697,11 +698,7 @@ public virtual bool IsPrefix(string source, string prefix, CompareOptions option
 
             if (prefix.Length == 0)
             {
-                return true;
-            }
-            if (source.Length == 0)
```

Comment:
we should keep this condition for ordinal cases.

Reply:
Fixed in most recent iteration.